test/Microsoft.AspNet.Mvc.Core.Test/Filters/FormatFilterTest.cs

```diff
@@ -0,0 +1,324 @@
+﻿using System;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Mvc;
+using Microsoft.AspNet.PipelineCore;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.OptionsModel;
+using Microsoft.Net.Http.Headers;
+using Xunit;
+
+#if ASPNET50
+using Moq;
+#endif
+
+namespace Microsoft.AspNet.Mvc.Test
+{
+    public class FormatFilterTests
+    {
+        public enum FormatSource
+        {
+            RouteData,
+            QueryData,
+            RouteAndQueryData
+        }
+
+        [Theory]
+        [InlineData("json", FormatSource.RouteData, "application/json")]
+        [InlineData("json", FormatSource.QueryData, "application/json")]
+        [InlineData("json", FormatSource.RouteAndQueryData, "application/json")]
+        public void FormatFilter_ContextContainsFormat_DefaultFormat(
+            string format, 
+            FormatSource place, 
+            string contentType)
+        {
+            // Arrange  
+            var mediaType = MediaTypeHeaderValue.Parse("application/json");
+            var resultExecutingContext = CreateResultExecutingContext(
+                format, 
+                FormatSource.RouteData);            
+            var resourceExecutingContext = CreateResourceExecutingContext(
+                new IFilter[] { }, 
+                format, 
+                FormatSource.RouteData);
+            var filter = new FormatFilterAttribute();
+
+            // Act
+            filter.OnResourceExecuting(resourceExecutingContext);
```

Comment:
Is there a result we should verify here? Maybe that it doesn't set `context.Result`?